GoogleForCreators/web-stories-wp
# Use SAT for detecting lasso collision with elements

## Summary
Improves lasso selection by using the Separating Axis Theorem for detecting if an element was inside the lasso selection or not.


## Relevant Technical Choices



## To-do



## User-facing changes
- Lasso selection now works correctly for rotated elements, too.


## Testing Instructions
1. Add a few elements, rotate some of those, too.
2. Now test out selecting different combinations of elements using lasso selection
3. Verify that the elements are selected correctly, e.g. even if only a corner of a rotated element stays inside the lasso selection.


---



Fixes #2215 
Fixes #1481
